lib/Find-VS2017.cs

```diff
@@ -0,0 +1,262 @@
+// powershell -ExecutionPolicy Unrestricted -Version "2.0" -Command "&{Add-Type -Path Find-VS2017.cs; [VisualStudioConfiguration.Main]::Query()}"
```

Comment:
This was submitted by @refack in https://github.com/nodejs/node-gyp/pull/1103 , I believe it was adapted from the python COM API by @bzoz in https://github.com/nodejs/node-gyp/pull/1101 , which in turn got the GUIDs and API from the nuget package upstream. @refack can you confim?